src/settings.rs

```diff
@@ -0,0 +1,775 @@
+//! A set of helpers to handle official DGC settings.
+
+use std::{borrow::Cow, fmt};
+
+use serde::{
+    de::{self, value::StrDeserializer, IntoDeserializer},
+    Deserialize, Serialize,
+};
+
+/// The URL from which the settings can be retrieved in JSON format.
+pub const URL: &str = "https://get.dgc.gov.it/v1/dgc/settings";
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Settings<'a> {
+    pub vaccines: Vaccines,
+    pub deny_list: DenyList<'a>,
+    pub min_versions: MinVersions<'a>,
+    pub tests: Tests,
+    pub recovery: Recovery,
+    pub unknown: Vec<RawSetting<'a>>,
+}
+
+#[derive(Debug, Default)]
+struct PartialSettings<'a> {
+    vaccines: PartialVaccines,
+    deny_list: Option<&'a str>,
+    min_versions: PartialMinVersions<'a>,
+    tests: PartialTests,
+    recovery: PartialRecovery,
+    unknown: Vec<RawSetting<'a>>,
+}
+
+impl<'a> PartialSettings<'a> {
+    fn get_field(&mut self, ty: SettingType, name: SettingName) -> Option<InnerField<'a, '_>> {
+        use SettingName::*;
+        use SettingType::*;
+
+        Some(match ty {
+            JensenVaccine | VaxzevriaVaccine | SpikevaxVaccine | ComirnatyVaccine
```

Comment:
Corbevax, Novavax, Variant-specific editions are caught before we hit this code?